packages/editor-ui/src/components/ExternalSecretsProviderModal.ee.vue

```diff
@@ -0,0 +1,331 @@
+<script lang="ts" setup>
+import Modal from './Modal.vue';
+import { EXTERNAL_SECRETS_PROVIDER_MODAL_KEY, MODAL_CONFIRM } from '@/constants';
+import { computed, onMounted, ref } from 'vue';
+import type { PropType, Ref } from 'vue';
+import type { EventBus } from 'n8n-design-system/utils';
+import { useExternalSecretsProvider, useI18n, useMessage, useToast } from '@/composables';
+import { useExternalSecretsStore } from '@/stores/externalSecrets.ee.store';
+import { useUIStore } from '@/stores';
+import { useRoute } from 'vue-router';
+import ParameterInputExpanded from '@/components/ParameterInputExpanded.vue';
+import type { IUpdateInformation, ExternalSecretsProviderData } from '@/Interface';
+import type { IParameterLabel } from 'n8n-workflow';
+import ExternalSecretsProviderImage from '@/components/ExternalSecretsProviderImage.ee.vue';
+import ExternalSecretsProviderConnectionSwitch from '@/components/ExternalSecretsProviderConnectionSwitch.ee.vue';
+import { createEventBus } from 'n8n-design-system/utils';
+import type { ExternalSecretsProvider } from '@/Interface';
+
+const props = defineProps({
+	data: {
+		type: Object as PropType<{ eventBus: EventBus; name: string }>,
+		default: () => ({}),
+	},
+});
+
+const defaultProviderData = {
+	infisical: {
+		siteURL: 'https://app.infisical.com',
+	},
```

Comment:
Requested by product. I didn't like it either, but it is what it is...

Reply:
Product requested that this be implemented on the FE? why cannot the backend return this?

Reply:
Because the default config is blank, and we're using the blank state of the config to determine the provider connection state. I suggested using it as a fallback, but there was push back for it.

Reply:
hmm okay